### PR TITLE
Generate Manpage from Docu

### LINKED ISSUE #143: Write/generate man page
Ideally we generate the man page from our markdown documentation.

### PR DESCRIPTION
**Pull request by Bloodchiefy**
Generated manpage with pandoc from README.md

Command used: `pandoc --standalone --to man README.md -o knoxite.1`

Open questions:
- [x] Shall the manpage be created from the README or the knoxite/website documentation?
- [x] Shall the generation be integrated into github workflows?
- [x] The manpage needs to be moved to `/usr/local/share/man/man1` as far as I know. So this needs to be done in an installed script for knoxite, right?
- [x] Clearify command usage documentation (`<` or `[`)
- [x] Check if `scripts` and `docs` pathnames are fine
- [x] Check if code is fine (remove probable uncommented code lines)

Closes #143

### CODE REVIEW

**Comment by muesli** [comment]
Just a heads up, here's yet another option we have: we're using `cobra` to define our command-line arguments and flags, which also comes with a documentation generator that builds the man page directly from the source code and command definitions.

**Comment by Bloodchiefy** [comment]
Alright, thanks, that helped.

**Comment by Bloodchiefy** [comment]
So, docs workflow is sadly not working yet, I call it another day. For the man pages generation with `cobra`, this helped: https://github.com/spf13/cobra/blob/master/doc/man_docs.md

I changed in the `Use` of each cobra command in knoxite, since the `<` and `>` symbols are getting `cobra` confused, thinking they may be html-tags. I will probably go back on that, and change those symbols with `&lt;` and `&gt;` to escape those symbols, hopefully the command line picks up on that. Otherwise, this discussion ment, usage of `[` and `]` is considered as best practice:
https://github.com/spf13/cobra/issues/1137

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  docs:
+    if: github.event.pull_request.head.repo.full_name == github.repository
+    runs-on: ubuntu-latest
+    steps:
+      - name: setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.13
```
Comment (muesli) [inline .github/workflows/docs.yml:13]: I'd remove this and let it automatically pick the latest Go version.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: docs
+
+on: [push, pull_request]
```
Comment (muesli) [inline .github/workflows/docs.yml:3]: We should probably only execute this on pushes to the master branch of the original repo.

File: .github/workflows/docs.yml (hunk 1 above)
Comment (Bloodchiefy) [reply .github/workflows/docs.yml]: removed `with`

File: .github/workflows/docs.yml (hunk 2 above)
Comment (Bloodchiefy) [reply .github/workflows/docs.yml]: changed to only running on master branch

**Comment by Bloodchiefy** [comment]
To prevent the generation to always auto-commit files, if only the date would change, I extended the Header-attribute of man pages:
```
t, _ := time.Parse(time.RFC822, "09 Aug 21 10:00 CEST")
header := &doc.GenManHeader{
		Title:   "KNOXITE",
		Section: "1",
		Date:    &t,
		Source:  "Auto generated by knoxite/knoxite",
	}
```
If the date needs to be changed, it has to be done **manually** here. The Source is set to `Auto generated by knoxite/knoxite`, I hope, that it is fine.

**Review by muesli — APPROVED** [review]
LGTM!

**Pull request merged**